ThingIFSDK/ThingIFSDKTests/PatchServerCodeTriggerTests.swift

```diff
@@ -103,13 +111,22 @@ class PatchServerCodeTriggerTests: SmallTestBase {
             }
         }
     }
-    
-    func testPatchServerCodeTrigger_http_404() {
+    func testPatchServerCodeStateTrigger_http_404() {
+        let condition = Condition(clause: EqualsClause(field: "color", intValue: 0))
+        let predicate = StatePredicate(condition: condition, triggersWhen: TriggersWhen.CONDITION_FALSE_TO_TRUE)
+        patchServerCodeTrigger_http_404(predicate)
+
+    }
+    func testPatchServerCodeScheduleOnceTrigger_http_404() {
+        let predicate = ScheduleOncePredicate(scheduleAt: NSDate(timeIntervalSinceNow: 1000))
+        patchServerCodeTrigger_http_404(predicate)
+    }
+    func patchServerCodeTrigger_http_404(predicate: Predicate) {
         let setting:TestSetting = TestSetting()
         let api = setting.api
         let tag = "PatchServerCodeTriggerTests.testPatchServerCodeTrigger_http_404"
         let triggerID = "0267251d9d60-1858-5e11-3dc3-00f3f0b5"
-        let expectation = self.expectationWithDescription("testPostNewServerCodeTrigger_http_404")
+        weak var expectation : XCTestExpectation! = self.expectationWithDescription("testPostNewServerCodeTrigger_http_404_\(predicate.getEventSource().rawValue)")
```

Comment:
please refer to https://github.com/apple/swift-corelibs-xctest/blob/eee815e96a068dded4b546e8693b65750ab552e9/Sources/XCTest/XCTestCase%2BAsynchronous.swift#L178
From the swift repo : 
- _allExpectations is an array which managed by xctest runtime
- Our testcases run parallel, in my opinion we should make all expectations `weak var`

additional information : 
- unlike regular class/subclass, testcase's `deinit` method is never called
- https://circleci.com/gh/KiiPlatform/thing-if-iOSSDK/356 shows that the test is timed out, this is absurd since it does not happen if run the test individually and the failure seems come from another unfullfilled expectations on another class.